lib/OperatingSystems/FreeBSD.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @author Matthew Wener <matthew@wener.org>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ServerInfo\OperatingSystems;
+
+/**
+ * Class FreeBSD
+ *
+ * @package OCA\ServerInfo\OperatingSystems
+ */
+class FreeBSD {
+	
+	/**
+	 * @return bool
+	 */
+	public function supported(): bool {
+		return false;
+	}
+
+	/**
+	 * Get memory will return a list key => value where all values are in bytes.
+	 * [MemTotal => 0, MemFree => 0, MemAvailable => 0, SwapTotal => 0, SwapFree => 0].
+	 *
+	 * @return array
+	 */
+	public function getMemory(): array {
+		$data = ['MemTotal' => -1, 'MemFree' => -1, 'MemAvailable' => -1, 'SwapTotal' => -1, 'SwapFree' => -1];
+
+		try {
+			$swapinfo = $this->executeCommand('/usr/sbin/swapinfo');
+		} catch (\RuntimeException $e) {
+			return $data;
+		}
+
+		$line = preg_split("/[\s]+/", $swapinfo);
+		if (count($line) > 3) {
+			$data['SwapTotal'] = (int)$line[3];
+			$data['SwapFree'] = $data['SwapTotal'] - (int)$line[2];
+		}
+
+		try {
+			$return = $this->executeCommand('/sbin/sysctl -n hw.physmem hw.pagesize vm.stats.vm.v_inactive_count vm.stats.vm.v_cache_count vm.stats.vm.v_free_count');
+			$return = preg_split('/\s+/', trim($return));
+
+			$data['MemTotal'] = (int)$return[0];
+			$data['MemAvailable'] = (int)$return[1] * ((int)$return[2] + (int)$return[3] + (int)$return[4]);
+		} catch (\RuntimeException $e) {
+			return $data;
+		}
+		return $data;
+	}
+
+	/**
+	 * Get name of the processor
+	 *
+	 * @return string
+	 */
+	public function getCPUName(): string {
+		try {
+			$data = $this->executeCommand('/sbin/sysctl -n hw.model');
+		} catch (\RuntimeException $e) {
+			return $data;
+		}
+		return $data;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getTime() {
+		try {
+			$uptime = $this->executeCommand('date');
+		} catch (\RuntimeException $e) {
+			return $uptime;
+		}
+		return $uptime;
+	}
+
+	/**
+	 * Get the total number of seconds the system has been up or -1 on failure.
+	 *
+	 * @return int
+	 */
+	public function getUptime(): int {
+		try {
+			$shell_boot = $this->executeCommand('/sbin/sysctl -n kern.boottime');
+			preg_match("/[\d]+/", $shell_boot, $boottime);
+			$time = $this->executeCommand('date +%s');
+			$uptimeInSeconds = (int)$time - (int)$boottime[0];
+		} catch (\RuntimeException $e) {
+			return $uptimeInSeconds;
```

Comment:
Please define the variable $uptimeInSeconds with a default value of -1 before the try block.

Reply:
Done